src/network/featureflags.h

```diff
@@ -1,103 +1,102 @@
-    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
-    THE SOFTWARE.
+   SlimeVR Code is placed under the MIT license
+   Copyright (c) 2023 SlimeVR Contributors
+
+   Permission is hereby granted, free of charge, to any person obtaining a copy
+   of this software and associated documentation files (the "Software"), to deal
+   in the Software without restriction, including without limitation the rights
+   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+   copies of the Software, and to permit persons to whom the Software is
+   furnished to do so, subject to the following conditions:
+
+   The above copyright notice and this permission notice shall be included in
+   all copies or substantial portions of the Software.
+
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+   THE SOFTWARE.
 */
 
 #ifndef SLIMEVR_FEATURE_FLAGS_H_
 #define SLIMEVR_FEATURE_FLAGS_H_
 
-#include <cstring>
 #include <algorithm>
-
-/**
- * Bit packed flags, enum values start with 0 and indicate which bit it is.
- *
- * Change the enums and `flagsEnabled` inside to extend.
-*/
-struct ServerFeatures {
-public:
-    enum EServerFeatureFlags: uint32_t {
-        // Server can parse bundle packets: `PACKET_BUNDLE` = 100 (0x64).
-        PROTOCOL_BUNDLE_SUPPORT,
-
-        // Add new flags here
-
-        BITS_TOTAL,
-    };
-
-    bool has(EServerFeatureFlags flag) {
-        uint32_t bit = static_cast<uint32_t>(flag);
-        return m_Available && (m_Flags[bit / 8] & (1 << (bit % 8)));
-    }
-
-    /**
-     * Whether the server supports the "feature flags" feature,
-     * set to true when we've received flags packet from the server.
-    */
-    bool isAvailable() {
-        return m_Available;
-    }
-
-    static ServerFeatures from(uint8_t* received, uint32_t length) {
-        ServerFeatures res;
-        res.m_Available = true;
-        memcpy(res.m_Flags, received, std::min(static_cast<uint32_t>(sizeof(res.m_Flags)), length));
-        return res;
-    }
-
-private:
-    bool m_Available = false;
-
-    uint8_t m_Flags[static_cast<uint32_t>(EServerFeatureFlags::BITS_TOTAL) / 8 + 1];
+#include <cinttypes>
+#include <cstring>
+#include <unordered_map>
+
+// I hate C++11 - they fixed this in C++14, but our compilers are old as the iceage
+struct EnumClassHash {
+	template <typename T>
```

Comment:
I do not like the template here. As we do not save any programming bytes with it. Also the code understanding is more difficult in my meaning. I think that's a possible improvement for the future.